pandas/core/construction.py

```diff
@@ -413,14 +413,7 @@ def sanitize_array(data, index, dtype=None, copy=False, raise_cast_failure=False
 
     elif isinstance(data, (list, tuple)) and len(data) > 0:
         if dtype is not None:
-            try:
-                subarr = _try_cast(data, dtype, copy, raise_cast_failure)
-            except Exception:
-                if raise_cast_failure:  # pragma: no cover
-                    raise
-                subarr = np.array(data, dtype=object, copy=copy)
-                subarr = lib.maybe_convert_objects(subarr)
-
+            subarr = _try_cast(data, dtype, copy, raise_cast_failure)
```

Comment:
How come we don't need this `try` anymore?

Reply:
We don't have any test cases that make it past here.  I'm planning to put some more effort into trying to think up a case that would, but if I/we can't, then this is unnecessary.